src/manifold/src/manifold.cpp

```diff
@@ -775,4 +778,61 @@ Manifold Manifold::TrimByPlane(glm::vec3 normal, float originOffset) const {
 }
 
 ExecutionParams& ManifoldParams() { return params; }
-}  // namespace manifold
\ No newline at end of file
+
+Manifold Manifold::Hull(const std::vector<glm::vec3>& pts,
+                        const std::vector<float>& props, const int numProp) {
+  const int numVert = pts.size();
+  // FIXME: Should there be a hull specific error?
+  if (numVert < 4) return Invalid();
+
+  std::vector<quickhull::Vector3<double>> vertices(numVert);
+  for (int i = 0; i < numVert; i++) {
+    vertices[i] = {pts[i].x, pts[i].y, pts[i].z};
+  }
+
+  quickhull::QuickHull<double> qh;
+  // bools: correct triangle winding, and use original indices
+  auto hull = qh.getConvexHull(vertices, false, true);
+  const auto& triangles = hull.getIndexBuffer();
+  const int numTris = triangles.size() / 3;
+
+  Mesh mesh;
+  Impl::MeshRelationD relation;
+  relation.numProp = numProp;
+  relation.properties = props;
+  mesh.vertPos = pts;
+
+  mesh.triVerts.reserve(numTris);
+  for (int i = 0; i < numTris; i++) {
+    const int j = i * 3;
+    mesh.triVerts.push_back({triangles[j], triangles[j + 1], triangles[j + 2]});
+  }
+  return Manifold(mesh);
+}
+
+Manifold Manifold::Hull() const {
+  auto mesh = GetMeshGL();
```

Comment:
So this is to get 1:1 properties to positions? Of course that means we'll get duplicate positions with different properties, so it's going to be basically random which properties end up in the output. Also, it'll be hard to make sense of them without an `originalID`, since some of those properties might mean different things. Actually it's worse that than: you could end up with a face interpolating UV coordinates into vertex colors or something, since there's no guarantee the verts are from the same input object. 

I wonder if we should just remove properties from the Hull entirely and do just geometry for this reason. I can't think of a way to make it sensical otherwise.

Reply:
I wasn't really sure myself since I don't know much about theesh properties business. I just did it this way to try and preserve properties as you said, hoping that you or pca could point out what more I should be preserving (or not). I did consider triangles ending up with weirdly disparate properties as a possibility, but I don't know/understand what is done with properties during booleans either.

Reply:
The nice thing about Booleans is we can guarantee a bunch of things: every result triangle is a subset of exactly one input triangle, so it will only every interpolate properties from that triangle. Our decimator is also aware, so it won't merge triangles that don't come from the same original triangle. 

Each triangle in a hull is either an input triangle, or a new triangle. I suppose we could try to maintain properties (and faceIDs) for input triangles, but we'd have no choice but to drop them for new triangles. Personally I think it sounds like more trouble than it's worth.

Reply:
Ok, sounds good :+1:. I've stripped out all of the properties stuff and just get the vertices via `GetMesh`.